src/public/js/util.js

```diff
@@ -38,6 +38,12 @@ var UTIL = (function() {
      0xb4, 0x7c, 0xb8, 0x26, 0x77, 0x99, 0xe3, 0xa5,
      0x67, 0x4a, 0xed, 0xde, 0xc5, 0x31, 0xfe, 0x18,
      0x0d, 0x63, 0x8c, 0x80, 0xc0, 0xf7, 0x70, 0x07]);
+  log3.get = function(x) {
```

Comment:
I am reading on inline caching right now. It is quiteinteresting and requires more understanding from me before discussing, but what I can say is that it is relevant to the discussion, but not actually what I was talking about caching before :P It does, however, potentially make some gets from arrays faster than others under the hood, so it is kinda relevant.
An effect inline caching will have for instance, is that the first call to exp3.get(x) is actually quite different than the first call to standardArray[x]. The second call however will be very similar due to inline caching. Essentially from what I gather, inline caching is kind of like the compiler internally making what would be like an object declaration in another language the first time it sees a data type it doesn't recognize. That way in the future when that object is encountered again, the compiler already knows what fields the object has instead of having to figure it out again (since it already did once). So since we are adding a field called get to exp3 for instance the compiler has to do extra work to figure out what get is and where in memory that functions code is stored. In the future though the compiler will already know because it cached that info. So the second call to exp3.get will be much faster because we don't need to look up what get is.

That is my current understanding, but I am finding it interesting so I am trying to read further but got a bunch of other stuff to do right now. I got assigned an ops ticket at work (like figuring out what broke on a customer system).